rpcs3/Emu/Cell/Modules/cellCamera.cpp

```diff
@@ -769,33 +710,21 @@ error_code cellCameraGetBufferSize(s32 dev_num, vm::ptr<CellCameraInfoEx> info)
 
 	auto& g_camera = g_fxo->get<camera_thread>();
 
-	if (!g_camera.init)
+	if (auto error = check_errors(g_camera.state))
 	{
-		return CELL_CAMERA_ERROR_NOT_INIT;
+		return error;
 	}
 
-	if (g_cfg.io.camera == camera_handler::null)
-	{
-		return not_an_error(CELL_CAMERA_ERROR_DEVICE_NOT_FOUND);
-	}
-
-	// the next few checks have a strange order, if I can trust the tests
-
-	if (!check_dev_num(dev_num))
+	if (!info)
 	{
 		return CELL_CAMERA_ERROR_PARAM;
 	}
 
-	if (g_camera.is_open)
+	if (g_camera.state >= camera_state::open)
 	{
 		return CELL_CAMERA_ERROR_ALREADY_OPEN;
 	}
 
-	if (!info)
```

Comment:
Changing the order of error checks is just wrong

Reply:
Fixed.